pkg/querier/tripperware/roundtrip_test.go

```diff
@@ -86,22 +91,49 @@ func TestRoundTrip(t *testing.T) {
 			return mockMiddleware{}
 		}),
 	}
-	tw := NewQueryTripperware(log.NewNopLogger(),
-		nil,
-		nil,
-		middlewares,
-		middlewares,
-		mockCodec{},
-		mockCodec{},
-	)
 
+	limits := validation.Limits{}
+	flagext.DefaultValues(&limits)
+	defaultOverrides, err := validation.NewOverrides(limits, nil)
+	require.NoError(t, err)
+
+	limitsWithVerticalSharding := validation.Limits{QueryVerticalShardSize: 3}
+	shardingOverrides, err := validation.NewOverrides(limitsWithVerticalSharding, nil)
+	require.NoError(t, err)
 	for _, tc := range []struct {
 		path, expectedBody string
+		limits             Limits
 	}{
-		{"/foo", "bar"},
-		{queryExemplar, "bar"},
-		{queryRange, responseBody},
-		{query, responseBody},
+		{
+			path:         "/foo",
+			expectedBody: "bar",
+			limits:       defaultOverrides,
+		},
+		{
+			path:         queryExemplar,
+			expectedBody: "bar",
+			limits:       defaultOverrides,
+		},
+		{
+			path:         queryRange,
+			expectedBody: responseBody,
+			limits:       defaultOverrides,
+		},
+		{
+			path:         query,
+			expectedBody: "bar",
+			limits:       defaultOverrides,
+		},
+		{
+			path:         queryNonShardable,
+			expectedBody: "bar",
+			limits:       defaultOverrides,
+		},
+		{
+			path:         query,
+			expectedBody: responseBody,
+			limits:       shardingOverrides,
+		},
```

Comment:
```suggestion
		},
		{
			path:         queryNonShardable,
			expectedBody: "bar",
			limits:       shardingOverrides,
		},
```
so roundtrip.go:156 is covered by the tests